### PR TITLE
Add # length shorthand #909

### PR DESCRIPTION
**Pull request by STRd6**
This add `x.#` as a shorthand for `x.length`. It doesn't support without the `.` because our current implementation of private identifiers doesn't support that yet either. We can add it for both at some point in the future.

See #909

### CODE REVIEW

File: source/parser.hera [hunk 1]
```diff
@@ -975,14 +975,27 @@ ThisLiteral
...
 
-# Adds a #id -> this.#id shorthand as a "PrivateThis" literal in most cases
+# Adds a #id -> this.#id shorthand as a "HashThis" literal in most cases
 # with the exception of keeping `#a in b` as is for branded in checks.
 # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#using_the_in_operator_to_implement_branded_checks
 # This is slightly different from how https://262.ecma-international.org/#sec-relational-operators handles relational operators
 # grammatically but should be equivalent in practice
-PrivateThis
+HashThis
+  LengthShorthand:id ( &( _+ ( ( Not __ In ) / In ) ) "" )?:beforeIn ->
```
Comment (edemaine) [inline source/parser.hera:986]: Is there a reason you had to add `""` here?

**Review by edemaine — APPROVED** [review]
Could you add documentation to `reference.md`? Or I can add it.

File: source/parser.hera (hunk 1 above)
Comment (STRd6) [reply source/parser.hera:986]: https://github.com/DanielXMoore/Hera/issues/16

Comment (edemaine) [reply source/parser.hera:986]: Ah, I see. Fun. I guess alternatively we could split into two rules, like `PrivateIdentifier`. No strong feeling either way.

Comment (STRd6) [reply source/parser.hera:986]: I generally prefer it as a single rule to prevent backtracking.

**Pull request merged**